client/src/Movies/Movie.js

```diff
@@ -26,42 +30,31 @@ export default class Movie extends Component {
       });
   };
   // Uncomment this code when you're ready for the stretch problems
-  // componentWillReceiveProps(newProps){
-  //   if(this.props.match.params.id !== newProps.match.params.id){
-  //     this.fetchMovie(newProps.match.params.id);
-  //   }
-  // }
+  componentDidUpdate(newProps) {
+    if (this.props.match.params.id !== newProps.match.params.id) {
+      this.fetchMovie(newProps.match.params.id);
+    }
+  }
 
-  // saveMovie = () => {
-  //   const addToSavedList = this.props.addToSavedList;
-  //   addToSavedList(this.state.movie)
-  // }
+  saveMovie = () => {
+    const addToSavedList = this.props.save;
+    addToSavedList(this.state.movie);
+  };
```

Comment:
I would have just done the following: 

```
saveMovie = () => {
    this.props.save(this.state.movie);
}
```

saves a little on space. the addToSavedList I think is an unneeded variable.